x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -109,6 +112,16 @@ Collection<Failure> verify(LogicalPlan plan, BitSet partialMetrics) {
         return failures.failures();
     }
 
+    private void checkLookupJoin(LogicalPlan p, Failures failures) {
```

Comment:
Drive by comment: `LookupJoin` could implement `PostAnalysisVerificationAware` and perform the check there.

Reply:
Thanks @bpintea, good point, doing it now